### ISSUE TITLE
kata-manager.sh assumes jq is installed

### ISSUE THREAD

**Issue by gabevenberg** [kata-containers/kata-containers#6270]
# problem
installing kata via kata-manager.sh will call `jq` without checking whether `jq` is installed. This can lead to a confusing error message implying that fetching from github itself failed.

# fix
add a more user friendly error message when jq is not found.

**Comment by jodh-intel** [comment]
Thanks for raising @gabevenberg. I agree - we can add an extra check and if not found explain how the user can install the package/command.

It might be getting a bit baroque but another approach could be to fall back to using `python` or `python3`, as these have  JSON support by default.

**Comment by jodh-intel** [comment]
Issue added to the outreach tasks ideas page: https://github.com/kata-containers/community/wiki/NDSU---Outreach-2023-work-item-ideas

**Comment by gabevenberg** [comment]
> Thanks for raising @gabevenberg. I agree - we can add an extra check and if not found explain how the user can install the package/command.
> 
> It might be getting a bit baroque but another approach could be to fall back to using `python` or `python3`, as these have JSON support by default.

This is true. From cursory examination, jq is not a default tool on ubuntu, debian, or fedora. Python3, however, is everywhere.

**Comment by gabevenberg** [comment]
> Thanks for raising @gabevenberg. I agree - we can add an extra check and if not found explain how the user can install the package/command.
> 
> It might be getting a bit baroque but another approach could be to fall back to using `python` or `python3`, as these have JSON support by default.

So, after re-implementing the jq stuff in python, I realized we are checking (and on some systems, even attempting to install) jq, but due to passing the -f flag to workaround #6259, the entire check gets bypassed. I can open a PR to remove the dependency on jq still, or we can just wait of #6259 to be fixed. (or do we want things like curl and jq, that are needed for the script to work at all, to be checked regardless of -f?)

**Comment by jodh-intel** [comment]
> but due to passing the -f flag to workaround https://github.com/kata-containers/kata-containers/issues/6259, the entire check gets bypassed.

@gabevenberg - looking at the code again, yes, another solution would be to simply call `check_deps()` before `pre_checks()` as that is really what the script should do:

```diff
diff --git a/utils/kata-manager.sh b/utils/kata-manager.sh
index 1cb65145f..aeda885a9 100755
--- a/utils/kata-manager.sh
+++ b/utils/kata-manager.sh
@@ -345,8 +345,8 @@ setup()

        [ "$force" = "true" ] && return 0

-       pre_checks
        check_deps
+       pre_checks
 }

 # Download the requested version of the specified project.
```

I'd be happy with any of these solutions, but it probably does make sense to start by fixing #6259.

**Comment by jodh-intel** [comment]
Closing as @gabevenberg resolved this issue on #6309.
